packages/squiggle-lang/src/fr/list.ts

```diff
@@ -255,7 +262,10 @@ export const library = [
       makeDefinition(
         [
           frArray(frGeneric("A")),
-          frLambdaTyped([frGeneric("A"), frOptional(frNumber)], frGeneric("B")),
+          frLambdaTyped(
+            [frGeneric("A"), frNamed("index", frOptional(frNumber))],
+            frGeneric("B")
+          ),
```

Comment:
Just a crazy thought: it'd be cool if we could write fr types as signatures, and `makeDefinition` would parse them from strings.
Like, `makeDefinition("((index: A) => B) => B[]", ...)`.

We'd have to parse the signature at runtime (slight startup overhead), and turn the signatures into the proper TypeScript type (this is a crazy part, but since https://github.com/codemix/ts-sql is possible, this might also be possible).

Reply:
Yea, I was curious about that too. Maybe we could eventually read it using Squiggle (maybe a simple version). It could be nice to use TS types, but my impression is that that wouldn't actually work.